project/.travis/before_install_test.sh.twig

```diff
@@ -6,9 +6,16 @@ if [ "${TRAVIS_PHP_VERSION}" != "hhvm" ]; then
     TRAVIS_INI_FILE="$PHP_INI_DIR/travis.ini"
     echo "memory_limit=3072M" >> "$TRAVIS_INI_FILE"
 
+    {% if 'mongodb' in services %}
     if [ "$TRAVIS_PHP_VERSION" '<' '7.0' ]; then
         echo "extension=mongo.so" >> "$TRAVIS_INI_FILE"
+    else
+        echo "extension=mongodb.so" >> "$TRAVIS_INI_FILE"
+
+        # Backwards compatibility with old mongo extension
+        composer require "alcaeus/mongo-php-adapter" --no-update
```

Comment:
I just tried it on SonataMediaBundle (affected by this PR), and here are the results:

https://travis-ci.org/sonata-project/SonataMediaBundle/builds/250476334?utm_source=github_status&utm_medium=notification

You can't install it on composer.json because it requires atleast PHP 5.6 and we are building with PHP 5.3, 5.4 and 5.5. There is no version of `alcaeus/mongo-php-adapter` compatible with PHP < 5.5.

If we don't want to require it like in this PR the only way is to assume we can't do it for 3.x branch and only do it for master, but IMO the are some reasons to be able to do it like this:

* Faster builds (PHP7 vs PHP5)
* SonataMongoDBAdminBundle is not building with PHP7 even if it has the requirement on composer.json
* We can drop this code once the old extension is not used anymore on any of our projects
* It does not hurt a lot because it is only being required if mongodb is defined as a service.